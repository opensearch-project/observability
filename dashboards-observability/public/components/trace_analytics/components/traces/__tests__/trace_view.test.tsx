/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import { TraceView } from '..';
import { coreStartMock } from '../../../../../../test/__mocks__/coreMocks';

describe('Trace view component', () => {
  configure({ adapter: new Adapter() });

  it('renders trace view', () => {
    const core = coreStartMock;
    const wrapper = shallow(
      <TraceView
        http={core.http}
        chrome={core.chrome}
        parentBreadcrumbs={[{ text: 'test', href: 'test#/' }]}
        traceId="test"
        mode="data_prepper"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
