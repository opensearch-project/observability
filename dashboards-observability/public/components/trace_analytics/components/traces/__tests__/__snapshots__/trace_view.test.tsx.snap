// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Trace view component renders trace view 1`] = `
<Fragment>
  <EuiPage>
    <EuiPageBody>
      <EuiFlexGroup
        alignItems="center"
        gutterSize="s"
      >
        <EuiFlexItem>
          <EuiTitle
            size="l"
          >
            <h2
              className="overview-content"
            >
              test
            </h2>
          </EuiTitle>
        </EuiFlexItem>
      </EuiFlexGroup>
      <EuiSpacer
        size="xl"
      />
      <EuiPanel>
        <PanelTitle
          title="Overview"
        />
        <EuiHorizontalRule
          margin="m"
        />
        <EuiFlexGroup>
          <EuiFlexItem>
            <EuiFlexGroup
              direction="column"
            >
              <EuiFlexItem
                grow={false}
              >
                <EuiText
                  className="overview-title"
                >
                  Trace ID
                </EuiText>
              </EuiFlexItem>
              <EuiFlexItem
                grow={false}
              >
                <EuiText
                  className="overview-title"
                >
                  Trace group name
                </EuiText>
                <EuiText
                  className="overview-content"
                  size="s"
                >
                  -
                </EuiText>
              </EuiFlexItem>
            </EuiFlexGroup>
          </EuiFlexItem>
          <EuiFlexItem>
            <EuiFlexGroup
              direction="column"
            >
              <EuiFlexItem
                grow={false}
              >
                <EuiText
                  className="overview-title"
                >
                  Latency
                </EuiText>
                <EuiText
                  className="overview-content"
                  size="s"
                />
              </EuiFlexItem>
              <EuiFlexItem
                grow={false}
              >
                <EuiText
                  className="overview-title"
                >
                  Last updated
                </EuiText>
                <EuiText
                  className="overview-content"
                  size="s"
                />
              </EuiFlexItem>
            </EuiFlexGroup>
          </EuiFlexItem>
          <EuiFlexItem>
            <EuiFlexGroup
              direction="column"
            >
              <EuiFlexItem
                grow={false}
              >
                <EuiText
                  className="overview-title"
                >
                  Errors
                </EuiText>
                <EuiText
                  className="overview-content"
                  size="s"
                >
                  -
                </EuiText>
              </EuiFlexItem>
            </EuiFlexGroup>
          </EuiFlexItem>
        </EuiFlexGroup>
      </EuiPanel>
      <EuiSpacer />
      <EuiFlexGroup>
        <EuiFlexItem
          grow={3}
        >
          <ServiceBreakdownPanel
            data={Array []}
          />
        </EuiFlexItem>
        <EuiFlexItem
          grow={7}
        >
          <SpanDetailPanel
            colorMap={Object {}}
            http={[MockFunction]}
            traceId="test"
          />
        </EuiFlexItem>
      </EuiFlexGroup>
      <EuiSpacer />
      <EuiPanel>
        <EuiFlexGroup>
          <EuiFlexItem>
            <PanelTitle
              title="Payload"
            />
          </EuiFlexItem>
        </EuiFlexGroup>
        <EuiHorizontalRule
          margin="m"
        />
      </EuiPanel>
    </EuiPageBody>
  </EuiPage>
</Fragment>
`;
